content/blog/understanding-the-dom/index.md

```diff
@@ -0,0 +1,519 @@
+
+For example, if you have [the style attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/style) associated to an element you're working with, you're able to read the values of the element:
+
+```html
+<!-- index.html -->
+<div style="background-color: green; color: white; width: 200px; height: 400px;" id="greenEl">
+  This element is green
+</div>
+```
+
+```javascript
+// index.js
+const greenElement = document.querySelector('#greenEl');
+console.log(greenElement.style.backgroundColor); // 'green'
+```
+
+![A screenshot of the element and the debugger console of the above code](green_style_element.png)
+
+Not only are you able to read the value in question, but you can write and edit them as well:
+
+```javascript
+greenElement.style.backgroundColor = 'red';
+```
+
+Will turn the element's background color red, for example.
+
+![The element has now turned the background red](red_style_element.png)
+
+Somewhat silly, seeing as how the `<div>` is no longer green. 🤭
+
+#### Limitations {#attribute-limitations}
+
+While attributes can be of great use to store data about an element, there's a limitation: Values are always stored as strings. This means that objects, arrays, and other non-string primitives must find a way to go to and from strings when being read and written.
+
+> While you've seen `style` attribute be read and written to by an object interface, if you inspect the element or use the `getAttribute` to access the attribute's HTML API value, you'll find that it's really a string with a pleasant API wrapped around it that lets you use an object to interface with the attribute value.
+>
+> ```javascript
+> console.log(mainTextElement.getAttribute('style')); // This will return a string value, despite the API that lets you use an object to read and write
+> ```
+>
+> The reasoning behind this incongruity is due to [the implicit mapping of the HTML API and the `Element` API as mentioned at the start of the previous section](#element-properties). The limitations described here will also apply to the HTML API of those types of properties.
+
+For example, we can [use `data` attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) in order to read and write values via attributes to any given element.
+
+```html
+<!-- index.html -->
+<ul id="list" data-listitems="2">
+  <li>List item 1</li>
+  <li>List item 2</li>
+</ul>
+```
+
+```javascript
+// index.js
+const listEl = document.querySelector('#list');
+console.log(listEl.dataset.listitems); // '2'
+listEl.dataset.listitems = 3;
+console.log(listEl.dataset.listitems); // '3'
+```
+
+![Demonstrating that dataset values are able to be read and written](list_dataset.png)
+
+If you'll notice, I wrote the string `'3'` instead of the numerical value `3` in the code sample's outputs in the comments despite using the numerical `3` to set the value. This behavior is due to how default non-string values are saved to attributes.
```

Comment:
```suggestion
Note that I wrote the string `'3'` instead of the numerical value `3` in the code sample's outputs in the comments despite using the numerical `3` to set the value. This behavior is due to how default non-string values are saved to attributes.
```